onadata/apps/api/tests/viewsets/test_open_data_viewset.py

```diff
@@ -110,6 +110,41 @@ def test_get_data_using_uuid(self):
         # cast generator response to list so that we can get the response count
         self.assertEqual(len(streaming_data(response)), 4)
 
+    def test_column_headers_match_row_heaaders(self):
+        """
+        Test that the row headers generated
+        match the column headers for the same form
+        """
+        self.view = OpenDataViewSet.as_view({
+            'get': 'schema'
+        })
+
+        _open_data = self.get_open_data_object()
+        uuid = _open_data.uuid
+        request1 = self.factory.get('/', **self.extra)
+        response1 = self.view(request1, uuid=uuid)
+        self.assertEqual(response1.status_code, 200)
+        self.assertListEqual(
+            ['column_headers', 'connection_name', 'table_alias'],
+            sorted(list(response1.data))
+        )
+
+        self._make_submissions()
+        self.view = OpenDataViewSet.as_view({
+            'get': 'data'
+        })
+        _open_data = self.get_open_data_object()
+        uuid = _open_data.uuid
+        request2 = self.factory.get('/', **self.extra)
+        response2 = self.view(request2, uuid=uuid)
+        self.assertEqual(response2.status_code, 200)
+        # cast generator response to list for easy manipulation
+        row_data = streaming_data(response2)
+        # assert that the length of the column_headers
+        # is equal to the row_headers randomly picked
+        self.assertEqual(
+            len(response1.data['column_headers']), len(row_data[3]))
+
```

Comment:
Would it be possible to compare actual fields to confirm that if you are expecting `'age', 'name', '_id'` you actually get those values? A length comparison only means you are expecting a specific length of the result, it does not ensure you are getting the data you intend to get.

Reply:
Sure, updated this to check for specific fields.

I initially was testing using length, in an attempt to make sure the same number of fields were present for both the row headers and the column headers